compiler/damlc/daml-prim-src/GHC/Classes.daml

```diff
@@ -365,3 +370,71 @@ False || x              =  x
 not                     : Bool -> Bool
 not True                =  False
 not False               =  True
+
+
+#ifdef DAML_NUMERIC
+
+-- | Is this a valid scale for the `Numeric` type?
+--
+-- This typeclass is used to prevent the creation of Numeric values
+-- with too large a scale. The scale controls the number of digits available
+-- after the decimal point, and it must be between 0 and 37 inclusive.
+--
+-- Thus the only available instances of this typeclass are `NumericScale 0`
+-- through `NumericScale 37`. This cannot be extended without additional
+-- compiler and runtime support. You cannot implement a custom instance
+-- of this typeclass.
+--
+-- If you have an error message in your code of the form "No instance for
+-- `(NumericScale n)`", this is probably caused by having a numeric literal
+-- whose scale cannot be inferred by the compiler. You can usually fix this
+-- by adding a type signature to the definition, or annotating the numeric
+-- literal directly (for example, instead of writing `3.14159` you can write
+-- `(3.14159 : Numeric 5)`).
+class NumericScale (n : Nat) where
+    -- | Get the scale of a `Numeric` as an integer. For example,
+    -- `numericScale (3.14159 : Numeric 5)` equals `5`.
+    numericScale : proxy n -> Int
+
+    -- | HIDE
+    numericScalePrivate : proxy n -> ()
+
+instance NumericScale 0 where numericScale _ = 0; numericScalePrivate _ = ()
```

Comment:
We could generate this with CPP :trollface:

Reply:
Looks like somebody is "growth hacking" on his contribution stats. :wink: